app/workers/functions.php

```diff
@@ -1,100 +1,315 @@
-    {
-        $this->executor = new Executor(App::getEnv('_APP_EXECUTOR_HOST'));
+        /** Update execution status */
+        $execution
+            ->setAttribute('status', $executionResponse['status'])
+            ->setAttribute('statusCode', $executionResponse['statusCode'])
+            ->setAttribute('response', $executionResponse['response'])
+            ->setAttribute('stdout', $executionResponse['stdout'])
+            ->setAttribute('stderr', $executionResponse['stderr'])
+            ->setAttribute('duration', $executionResponse['duration']);
+    } catch (\Throwable $th) {
+        $interval = (new \DateTime())->diff(new \DateTime($execution->getCreatedAt()));
+        $execution
+            ->setAttribute('duration', (float)$interval->format('%s.%f'))
+            ->setAttribute('status', 'failed')
+            ->setAttribute('statusCode', $th->getCode())
+            ->setAttribute('stderr', $th->getMessage());
+        Console::error($th->getMessage());
     }
 
-    public function run(): void
-    {
-        $type = $this->args['type'] ?? '';
-        $events = $this->args['events'] ?? [];
-        $project = new Document($this->args['project'] ?? []);
-        $user = new Document($this->args['user'] ?? []);
-        $payload = json_encode($this->args['payload'] ?? []);
+    $execution = $dbForProject->updateDocument('executions', $executionId, $execution);
+
+    /** Trigger Webhook */
+    $executionModel = new Execution();
+    $executionUpdate = new Event(Event::WEBHOOK_QUEUE_NAME, Event::WEBHOOK_CLASS_NAME);
+    $executionUpdate
+        ->setProject($project)
+        ->setUser($user)
+        ->setEvent('functions.[functionId].executions.[executionId].update')
+        ->setParam('functionId', $function->getId())
+        ->setParam('executionId', $execution->getId())
+        ->setPayload($execution->getArrayCopy(array_keys($executionModel->getRules())))
+        ->trigger();
+
+    /** Trigger Functions */
+    $executionUpdate
+        ->setClass(Event::FUNCTIONS_CLASS_NAME)
+        ->setQueue(Event::FUNCTIONS_QUEUE_NAME)
+        ->trigger();
+
+    /** Trigger realtime event */
+    $allEvents = Event::generateEvents('functions.[functionId].executions.[executionId].update', [
+        'functionId' => $function->getId(),
+        'executionId' => $execution->getId()
+    ]);
+    $target = Realtime::fromPayload(
+    // Pass first, most verbose event pattern
+        event: $allEvents[0],
+        payload: $execution
+    );
+    Realtime::send(
+        projectId: 'console',
+        payload: $execution->getArrayCopy(),
+        events: $allEvents,
+        channels: $target['channels'],
+        roles: $target['roles']
+    );
+    Realtime::send(
+        projectId: $project->getId(),
+        payload: $execution->getArrayCopy(),
+        events: $allEvents,
+        channels: $target['channels'],
+        roles: $target['roles']
+    );
+
+    /** Update usage stats */
+    global $register;
```

Comment:
Avoid the global var. We can also set statsd as a dependency injection resource.

Reply:
Avoided global register.